esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -6118,7 +6104,289 @@ bool CWsDfuEx::onDFUFileAccess(IEspContext &context, IEspDFUFileAccessRequest &r
             userDesc.setown(createUserDescriptor());
             userDesc->set(userID.str(), context.queryPassword(), context.querySignature());
         }
-        getFileAccess(context, userDesc, accessType, req, resp);
+        getFileAccess(context, userDesc, accessType, requests.getName(), requests.getJobId(), requests, resp.updateAccessInfo());
+    }
+    catch (IException *e)
+    {
+        FORWARDEXCEPTION(context, e,  ECLWATCH_INTERNAL_ERROR);
+    }
+    return true;
+}
+
+const char *CWsDfuEx::createWsDFUCreateGroupName(StringArray &locations, StringBuffer &groupName)
+{
+    if (locations.length() < 1)
+        return nullptr;
+
+    StringBuffer buf;
+    ForEachItemIn(i, locations)
+        buf.append(locations.item(i)).append(";");
+    groupName.set("wsdfucreate").append(hashc((unsigned char*) buf.str(), buf.length(), 0));
+    return groupName.str();
+}
+
+IGroup *CWsDfuEx::queryDFUFileIGroup(StringBuffer &groupName, ClusterType clusterType, StringArray &locations)
+{
+    GroupType groupType;
+    StringBuffer basedir;
+    Owned<IGroup> group = queryNamedGroupStore().lookup(groupName.str(), basedir, groupType);
```

Comment:
What I expected this code would do, was to create an IGroup.
Then look it up and see if there is an existing matching layout.
```
IGroup *myProspectiveNewGroup = <create it>
You can use INamedGroupStore::find(myProspectiveNewGroup, true)
```

.. for that.

Then you can name it what you like (don't need hashed locations).

Reply:
The code here checks the groupName. If the groupName exists, I use the existing one.

Reply:
On reflection, INamedGroupStore::find() approach may be a bit expensive, as it must scan and check all groups to see if there's a match.
So creating a name based on hash of locations and checking to see if that exists 1st is a good approach.
But you should check if you get a match that the group really is the same, if it is not, you'll need to extend the group name and publish the new group.

Reply:
How to check that the group really is the same?  If the name (wsdfucreate + hash(IPs)) matches, the hash(IPs) will be the same.

Reply:
Have a look at the IGroup methods, there's an equals() method there.